rai/node/bootstrap.cpp

```diff
@@ -351,14 +200,22 @@ void rai::frontier_req_client::receive_frontier ()
 	});
 }
 
-void rai::frontier_req_client::unsynced (MDB_txn * transaction_a, rai::block_hash const & ours_a, rai::block_hash const & theirs_a)
+void rai::frontier_req_client::unsynced (MDB_txn * transaction_a, rai::block_hash const & head, rai::block_hash const & end)
 {
-	auto current (ours_a);
-	while (!current.is_zero () && current != theirs_a)
+	if (bulk_push_cost < bulk_push_cost_limit)
 	{
-		connection->node->store.unsynced_put (transaction_a, current);
-		auto block (connection->node->store.block_get (transaction_a, current));
-		current = block->previous ();
+		connection->attempt->add_bulk_push_target (head, end);
+		if (!connection->node->wallets.exists (transaction_a, current))
```

Comment:
I think this check can be removed, it seems like it's checking if a block hash exists as an account in any of the wallets which will never be true.

Reply:
Yeah that's a mistake. I'll probably pass in the account as a parameter to check this.

Reply:
Actually this was correct. It was using `current` not `head`. It shouldn't be too important though.